src/app/api/user/[uid]/edit-profile/route.ts

```diff
@@ -0,0 +1,112 @@
+import { NextRequest, NextResponse } from "next/server";
+import { EditProfileRequest, EditPositionRequest } from "./route.schema";
+import { EditProfileResponse } from "./route.schema";
+import { prisma } from "@server/db/client";
+import { withSession } from "@server/decorator";
+
+/* PATCH Request for updating profile information */
+export const PATCH = async (
+  request: NextRequest,
+  { params }: { params: { uid: string } }
```

Comment:
We can simplify this logic here a bit. Instead of defining the params in the function signature, simply export a function that's wrapped in `withSession(...)`

```jsx
export const PATCH = withSession(async () => { /* define your make patch logic here */ });
```